src/OpenLoco/src/Windows/ObjectSelectionWindow.cpp

```diff
@@ -249,28 +265,74 @@ namespace OpenLoco::Ui::Windows::ObjectSelectionWindow
         }
 
         // Add a marker to denote the last tab
-        _tabInformation[tabPos].index = 0xFF;
+        _tabPositions[tabPos].index = 0xFF;
 
-        const auto firstTabIndex = ObjectType(_tabInformation[0].index);
+        const auto firstTabIndex = ObjectType(_tabPositions[0].index);
         repositionTargetTab(self, firstTabIndex);
     }
 
-    // 0x00472BBC
-    static ObjectManager::ObjIndexPair sub_472BBC(Window* self)
+    static bool contains(const std::string_view& a, const std::string_view& b)
     {
-        const auto objects = ObjectManager::getAvailableObjects(static_cast<ObjectType>(self->currentTab));
+        return std::search(a.begin(), a.end(), b.begin(), b.end(), [](char a, char b) {
+                   return tolower(a) == tolower(b);
+               })
+            != a.end();
+    }
 
+    static void applyFilterToObjectList()
+    {
+        std::string_view pattern = inputSession.buffer.c_str();
```

Comment:
just construct it directly and avoid having to do a strlen on the buffer when we know its size.

Reply:
Right, I had forgotten the buffer is already a std::string. This would do, right?
```suggestion
        std::string_view pattern = inputSession.buffer;
```